src/components/query/PropertyQueryEditor.tsx

```diff
@@ -258,6 +314,35 @@ export class PropertyQueryEditor extends PureComponent<Props, State> {
     );
   }
 
+  renderOptions(query: SitewiseQuery, showProp: boolean, showQuality: boolean) {
```

Comment:
maybe let's break this up into a functional component rather than building another additional render function on this class? I think smaller functional components tend to be a bit easier to read and can even have some performance boosts